tests/integration/extension/src/background.js

```diff
@@ -1,15 +1,52 @@
 /* eslint-disable no-unused-vars */
 import * as webScience from "@mozilla/web-science"
+import browser from "webextension-polyfill";
+
+async function pageVisitStartListener(pageVisit) {
+    const tabs = await browser.tabs.query({
+        currentWindow: true,
+        active: true
+    });
+
+    for (let tab of tabs) {
+        browser.tabs.sendMessage(
+            tab.id,
+            { message: `WebScienceTest - Page visit start: ${JSON.stringify(pageVisit)}` }
+        );
+    }
 
-function pageVisitStartListener(pageVisit) {
     console.debug("WebScienceTest - Page visit start:", pageVisit);
 }
 
-function pageVisitStopListener(pageVisit) {
+async function pageVisitStopListener(pageVisit) {
+    const tabs = await browser.tabs.query({
+        currentWindow: true,
+        active: true
+    });
+
+    for (let tab of tabs) {
+        browser.tabs.sendMessage(
+            tab.id,
+            { message: `WebScienceTest - Page visit stop: ${JSON.stringify(pageVisit)}` }
+        );
+    }
+
     console.debug("WebScienceTest - Page visit stop:", pageVisit);
 }
 
-function pageDataListener(pageData) {
+async function pageDataListener(pageData) {
+    const tabs = await browser.tabs.query({
+        currentWindow: true,
+        active: true
+    });
+
+    for (let tab of tabs) {
```

Comment:
We can actually refactor a method that loops and sends a message to all tabs. Then call it from here and above two places.

Reply:
I'm not sure this is actually working as intended anyway. I think what we might need to do for Chrome is have a window we keep open that handles receiving these messages (this is similar to how the Firefox integration tests work).

Reply:
Yes I agree, I removed this for now since Chrome support needs a little more work and I want to split it out to a separate PR, and focus on getting the infrastructure. Getting log messages out of extensions is a real hassle, I want to research it a little more.